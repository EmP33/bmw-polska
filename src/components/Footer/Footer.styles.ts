import styled from "styled-components";

export const Container = styled.footer`
  width: 100%;
  min-height: 30vh;
  background-color: var(--color-gray-7);
  display: grid;
  font-family: var(--font-secondary);

  .wrapper {
    width: 75%;
    margin: 0 auto;

    @media only screen and (max-width: 1200px) {
      width: 85%;
    }
    @media only screen and (max-width: 900px) {
      width: 95%;
    }
    @media only screen and (max-width: 650px) {
      width: 85%;
    }
  }

  .socials {
    width: 100%;
    border-bottom: 1px solid var(--color-gray-6);
    padding: 1.5rem 0 1.5rem 0;
    display: grid;
    grid-template-columns: repeat(2, max-content);
    align-items: center;
    justify-content: end;
    column-gap: 1.75rem;

    @media only screen and (max-width: 650px) {
      grid-template-rows: repeat(2, 1fr);
      grid-template-columns: max-content;
      justify-content: start;
      row-gap: 2rem;
      padding: 4rem 0 1.5rem 0;
    }

    span {
      font-size: 1.25rem;
      color: var(--color-gray-3);
      font-weight: 300;
      display: block;
    }

    &__icons {
      display: grid;
      align-items: grid;
      column-gap: 1.75rem;

      svg {
        display: block;
        width: 1.25rem;
        height: 1.25rem;
        fill: #333;
        cursor: pointer;
        grid-row: 2;

        &:hover {
          fill: #222;
        }
      }
    }
  }

  .footer-links {
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    padding: 3rem 0 5rem 0;

    @media only screen and (max-width: 650px) {
      grid-template-columns: 1fr;
      row-gap: 3rem;
    }

    & > div {
      span {
        color: var(--color-gray-3);
        font-size: 1.25rem;
        font-weight: 300;
        display: block;
        margin-bottom: 2rem;
      }

      ul {
        list-style: none;
        display: grid;
        row-gap: 0.5rem;

        a {
          color: var(--color-gray-3);
          text-decoration: none;
          font-weight: bold;
          font-size: 0.875rem;

          @media only screen and (max-width: 650px) {
            font-size: 0.75rem;
          }

          &:hover {
            color: #333;
          }
        }
      }
    }
  }

  .copyright-wrapper {
    background-color: var(--color-white);
    .copyright {
      width: 75%;
      margin: 0 auto;
      padding: 1.5rem 0 1.5rem 0;
      font-size: 0.875rem;
      font-family: var(--font-primary);
      color: var(--color-gray-1);
      @media only screen and (max-width: 1200px) {
        width: 85%;
      }
      @media only screen and (max-width: 900px) {
        width: 95%;
      }
      @media only screen and (max-width: 650px) {
        text-align: center;
      }
    }
  }
`;
