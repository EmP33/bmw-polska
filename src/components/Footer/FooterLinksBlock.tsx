import React from "react";

interface Props {
  title: string;
  elements: string[];
}

const FooterLinksBlock: React.FC<Props> = ({ title, elements }) => {
  return (
    <div className="footer-links__block">
      <span>{title}</span>
      <ul>
        {elements.map((element) => (
          <li>
            <a href="#">{element}</a>
          </li>
        ))}
      </ul>
    </div>
  );
};

export default FooterLinksBlock;
