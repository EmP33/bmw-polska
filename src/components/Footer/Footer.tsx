import React from "react";
// Styles
import { Container } from "./Footer.styles";
// Icons
import {
  FacebookIcon,
  InstagramIcon,
  YoutubeIcon,
  LinkedinIcon,
  TwitterIcon,
} from "../../assets";
// Components
import FooterLinksBlock from "./FooterLinksBlock";

const Footer = () => {
  return (
    <Container>
      <div className="wrapper">
        <div className="socials">
          <span>Dołącz do nas</span>
          <div className="socials__icons">
            <FacebookIcon />
            <InstagramIcon />
            <YoutubeIcon />
            <LinkedinIcon />
            <TwitterIcon />
          </div>
        </div>
        <div className="footer-links">
          <FooterLinksBlock
            title="Strefa Klienta"
            elements={[
              "Centrum Obsługi Klienta BMW",
              "Znajdź dealera BMW",
              "Pobierz cenniki / broszury",
              "Często zadawane pytania",
            ]}
          />
          <FooterLinksBlock
            title="Poznaj BMW"
            elements={["Informacje prasowe", "BMW Polska", "Kariera"]}
          />
          <FooterLinksBlock
            title="Informacje prawne"
            elements={[
              "Mapa serwisu",
              "Nota prawna / Prywatność",
              "Polityka cookie",
              "Warunki korzystania z MyBMW",
            ]}
          />
        </div>
      </div>
      <div className="copyright-wrapper">
        <div className="copyright">&copy; BMW Polska 2020</div>
      </div>
    </Container>
  );
};

export default Footer;
