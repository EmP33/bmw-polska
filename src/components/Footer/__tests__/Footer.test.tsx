import Footer from "../Footer";
import ReactShallowRenderer from "react-test-renderer/shallow";

describe("<Footer />", () => {
  it("should render Footer properly", () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<Footer />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
