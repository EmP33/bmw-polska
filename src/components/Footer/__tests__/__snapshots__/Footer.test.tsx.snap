// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render Footer properly 1`] = `
<styled.footer>
  <div
    className="wrapper"
  >
    <div
      className="socials"
    >
      <span>
        Dołącz do nas
      </span>
      <div
        className="socials__icons"
      >
        <FacebookIcon />
        <InstagramIcon />
        <YoutubeIcon />
        <LinkedinIcon />
        <TwitterIcon />
      </div>
    </div>
    <div
      className="footer-links"
    >
      <FooterLinksBlock
        elements={
          Array [
            "Centrum Obsługi Klienta BMW",
            "Znajdź dealera BMW",
            "Pobierz cenniki / broszury",
            "Często zadawane pytania",
          ]
        }
        title="Strefa Klienta"
      />
      <FooterLinksBlock
        elements={
          Array [
            "Informacje prasowe",
            "BMW Polska",
            "Kariera",
          ]
        }
        title="Poznaj BMW"
      />
      <FooterLinksBlock
        elements={
          Array [
            "Mapa serwisu",
            "Nota prawna / Prywatność",
            "Polityka cookie",
            "Warunki korzystania z MyBMW",
          ]
        }
        title="Informacje prawne"
      />
    </div>
  </div>
  <div
    className="copyright-wrapper"
  >
    <div
      className="copyright"
    >
      © BMW Polska 2020
    </div>
  </div>
</styled.footer>
`;
